ibis/expr/api.py

```diff
@@ -4193,27 +4193,60 @@ def mutate(table, exprs=None, **mutations):
         for name, expr in sorted(mutations.items(), key=operator.itemgetter(0))
     )
 
+    selection_exprs_by_name = collections.OrderedDict()
     for expr in exprs:
-        if expr.get_name() and isinstance(expr, ir.DestructColumn):
-            raise com.ExpressionError(
-                f"Cannot name a destruct column: {expr.get_name()}"
-            )
+        if isinstance(expr, ir.DestructColumn):
+            if expr.get_name():
+                raise com.ExpressionError(
```

Comment:
test hits this?

Reply:
Yes tested already here: https://github.com/ibis-project/ibis/blob/master/ibis/backends/tests/test_vectorized_udf.py#L253